### ISSUE TITLE
Can't fit for the secondary radius when map has reflected light

### ISSUE THREAD

**Issue by christinahedges** [rodluger/starry#240]
**Describe the bug**
I'm trying to fit a system with reflected light turned on. I want to fit for the radius at the same time. This works great with reflected light off, but when reflected is on the secondary radius seems to throw a theano error.

**To Reproduce**
```python
import starry
import pymc3 as pm
import numpy as np
import exoplanet as xo
starry.config.lazy = True
starry.config.quiet = True

t, y, yerr = np.arange(1000), np.ones(1000) + np.random.normal(0, 0.01, 1000), np.ones(1000) * 0.01

with pm.Model() as model:
    
    # Secondary
    B_r = pm.Uniform("B_r", lower=0.005, upper=0.05, testval=0.02)

    
    sec_map = starry.Map(ydeg=1, reflected=True, amp=0.01)
    ncoeff = sec_map.Ny - 1
    sec_mu = np.zeros(ncoeff)
    sec_cov = 1e-2 * np.eye(ncoeff)
    map_coefficients = pm.MvNormal("sec_y", sec_mu, sec_cov, shape=(ncoeff,)) 
    
    sec_map[1:, :] = map_coefficients
    sec = starry.Secondary(sec_map,
        r=B_r,
        porb=10,
    )
    
    # Primary
    pri_map = starry.Map(udeg=1, amp=1)
    pri = starry.Primary(
        pri_map,
        r=1)

    sys = starry.System(pri, sec)
    
    
    pm.Normal("obs", mu=sys.flux(t=t), sd=yerr**2, observed=y) 
    map_soln = xo.optimize(start=model.test_point, verbose=False)
```
**Expected behavior**

I expected that this would produce a map_soln with some values. 

**Your setup (please complete the following information):**
 - Version of starry: 1.0.1.dev49+g871582f6
 - Operating system: Mac OS Mojave
 - Python version & installation method (pip, conda, etc.): github

**Additional context**
This is the error message I get:

```python

ValueError: expected an ndarray
Apply node that caused the error: Elemwise{Composite{((-i0) / sqr(i1))}}[(0, 0)](<starry._core.ops.integration.sTReflectedGradientOp object at 0x1478bd4e0>.3, Subtensor{int64}.0)
Toposort index: 172
Inputs types: [TensorType(float64, scalar), TensorType(float64, scalar)]
Inputs shapes: [(), ()]
Inputs strides: [(), ()]
Inputs values: [0.0, array(0.02)]
Outputs clients: [[IncSubtensor{InplaceInc;int64}(IncSubtensor{InplaceInc;int64}.0, Elemwise{Composite{((-i0) / sqr(i1))}}[(0, 0)].0, Constant{0})]]

HINT: Re-running with most Theano optimization disabled could give you a back-trace of when this node was created. This can be done with by setting the Theano flag 'optimizer=fast_compile'. If that does not work, Theano optimizations can be disabled with 'optimizer=None'.
HINT: Use the Theano flag 'exception_verbosity=high' for a debugprint and storage map footprint of this apply node.
```

**Comment by rodluger** [comment]
Thanks for reporting this! I pushed a simple shape fix that seems to have done the trick. Can you check that it works (and that you get sensible output when you apply this to an actual dataset)? Thanks!

**Comment by rodluger** [comment]
FYI I just found a separate bug affecting the depth of secondary eclipse in reflected light (you'll probably run into it -- it's quite bad). Working on fixing that now...

**Comment by rodluger** [comment]
Fixed #241.

**Comment by christinahedges** [comment]
Yep this fixes the issue! Thanks for the fast response!
